### PR TITLE
Fix DML RETURNING data silently discarded on wire protocol

### PR DESCRIPTION
**Pull request by jghoman**
## Summary

- INSERT/UPDATE/DELETE with RETURNING clauses were routed through `Exec` (which discards result rows) because the query router checked `cmdType != "SELECT"` instead of whether the query actually returns results
- All 7 routing paths now use `queryReturnsResults()` to decide between Exec and Query
- Correct command tags emitted (`INSERT 0 N`, `UPDATE N`, `DELETE N`)
- Describe skips schema probe for DML to prevent double-execution
- Parenthesized queries (e.g. `(SELECT ... UNION ...)`) now route correctly

| Protocol path | Before | After |
|---|---|---|
| Simple query (psql) | RETURNING data silently lost | **Fixed.** Correct data + tags |
| Extended - client skips Describe | RETURNING data silently lost | **Fixed.** Correct data + tags |
| Extended - client sends Describe (lib/pq) | NoData, data silently lost | NoData, explicit error — **no silent data loss** |

## Test plan

- [x] 51 unit tests for `queryReturnsResults()` (RETURNING detection, case sensitivity, multi-line, false positive resistance, parenthesized queries)
- [x] 17 unit tests for `buildCommandTagFromRowCount()` (all DML types, edge cases)
- [x] 12 consistency tests verifying Exec and Query paths produce identical command tags
- [x] Full `go test ./...` passes (only pre-existing `psql_dn` integration failure)
- [x] End-to-end smoke test via psql: INSERT/UPDATE/DELETE RETURNING all return correct data and tags

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -1143,7 +1143,8 @@ func (c *clientConn) executeSelectQuery(query string) error {
 		return nil
 	}
 
-	tag := fmt.Sprintf("SELECT %d", rowCount)
+	c.updateTxStatus(cmdType)
+	tag := buildCommandTagFromRowCount(cmdType, int64(rowCount))
```
[the remark this answers is not in the document]
Comment (jghoman) [reply server/conn.go]: Fixed.
